code/game/objects/items/devices/spyglasses.dm

```diff
@@ -2,13 +2,13 @@
 /obj/item/clothing/glasses/sunglasses/spy
 	desc = "Made by Nerd. Co's infiltration and surveillance department. Upon closer inspection, there's a small screen in each lens."
 	actions_types = list(/datum/action/item_action/activate_remote_view)
-	var/obj/item/clothing/accessory/spy_bug/linked_bug
+	var/datum/component/spybug/linked_bug
```

Comment:
I don't think this is how you handle and add components to items

Reply:
The spybug component is attached to the spybug (the pen or the shuriken), not the glasses, and this is used to access the remote bug component on the separate item.

Reply:
> I don't think this is how you handle and add components to items

This is most definitely not.

Please review other component implementations.